scheduler/core/components/collector/collector.py

```diff
@@ -301,138 +303,152 @@ def _calculate_target_info(self,
-                elev_min = obs.constraints.elevation_min
-                elev_max = obs.constraints.elevation_max
+            # Grab redis key to check in cache
+            key = f'{obs.id.id}{jday}{Collector.DEFAULT_TIMESLOT_LENGTH.value}'
+            logger.info(f'Checking redis key {key}')
+            exists = redis_client.exists(key)
+            if exists:
+                logger.info(f'Retrieving from Redis key {key}')
+                serialized_ti = json.loads(redis_client.get(key))
+                ti = TargetInfo.from_dict(serialized_ti)
             else:
-                targ_prop = airmass
-                elev_min = Constraints.DEFAULT_AIRMASS_ELEVATION_MIN
-                elev_max = Constraints.DEFAULT_AIRMASS_ELEVATION_MAX
-
-            # Are all the required resources available?
-            # This works for validation mode. In RT mode, this may need to be statistical if resources are not known
-            # and they could change with time, so the visfrac calc may need to be extracted from this method
-            has_resources = all([resource in nc[night_idx].resources for resource in obs.required_resources()])
-            avail_resources = np.full([len(night_events.times[night_idx])], int(has_resources), dtype=int)
-
-            # Is the program excluded on a given night due to block scheduling
-            prog = self.get_program(obs.id.program_id())
-            can_schedule = nc[night_idx].filter.program_filter(prog)
-            is_schedulable = np.full([len(night_events.times[night_idx])], int(can_schedule), dtype=int)
-            # print(f"{obs.unique_id} {has_resources} {can_schedule}")
-
-            # Calculate the time slot indices for the night where:
-            # 1. The sun altitude requirement is met (precalculated in night_events)
-            # 2. The sky background constraint is met
-            # 3. The elevation constraints are met
-            sa_idx = night_events.sun_alt_indices[night_idx]
-
-            c_idx = np.where(
-                np.logical_and(sb[sa_idx] <= targ_sb,
-                               np.logical_and(avail_resources[sa_idx] == 1,
-                                              np.logical_and(is_schedulable[sa_idx] == 1,
-                                                             np.logical_and(targ_prop[sa_idx] >= elev_min,
-                                                                            targ_prop[sa_idx] <= elev_max))))
-            )[0]
-
-            # Apply timing window constraints.
-            # We always have at least one timing window. If one was not given, the program length will be used.
-            for tw in timing_windows:
-                tw_idx = np.where(
-                    np.logical_and(night_events.times[night_idx][sa_idx[c_idx]] >= tw[0],
-                                   night_events.times[night_idx][sa_idx[c_idx]] <= tw[1])
+                # Calculate the ra and dec for each target.
```

Comment:
GitHub says the code starting here has changed considerably, but it's going to take awhile to manually go through and see what changed given how much is indicated. Could you give a brief summary or what the changes are in this PR?